src/qtils/hex.hpp

```diff
@@ -17,23 +17,21 @@ namespace qtils {
    * @struct Hex
    * @brief Wrapper for hex encoding of byte sequences.
    */
-  struct Hex {
-    SpanAdl<const uint8_t> v;  ///< Underlying byte sequence to be hex-encoded
-  };
+  struct Hex : std::span<const uint8_t> {};
 
   /**
-   * @brief Convert byte container wrapped in Hex to hexadecimal string.
+   * @brief Convert a byte container wrapped in Hex to hexadecimal string.
    * @param data Hex wrapper around BytesIn
    * @return Uppercase hex-encoded string (2 chars per byte)
    */
   inline std::string to_hex(const Hex &data) {
```

Comment:
There should be an overload which takes the destination as some kind of pointer or iterator to avoid allocations where the output size is small and bounded at compile time.

Reply:
I though about remove it, because `fmt::format` is more flexible and completely replaces that.

Reply:
removed